cmd/juju/controller/killstatus.go

```diff
@@ -58,25 +60,26 @@ func newTimedStatusUpdater(ctx *cmd.Context, api destroyControllerAPI, controlle
 
 		// If we hit an error, status.HostedModelCount will be 0, the polling
 		// loop will stop and we'll go directly to destroying the model.
-		ctrStatus, modelsStatus, err := newData(api, controllerModelUUID)
+		ctrStatus, modelsStatus, applications, err := newData(api, controllerModelUUID)
 		if err != nil {
 			ctx.Infof("Unable to get the controller summary from the API: %s.", err)
 		}
 
 		return environmentStatus{
-			controller: ctrStatus,
-			models:     modelsStatus,
+			controller:   ctrStatus,
+			models:       modelsStatus,
+			applications: applications,
 		}
 	}
 }
 
-func newData(api destroyControllerAPI, controllerModelUUID string) (ctrData, []modelData, error) {
+func newData(api destroyControllerAPI, controllerModelUUID string) (ctrData, []modelData, []base.Application, error) {
```

Comment:
Perhaps this should now just return `environmentStatus` (+ error) instead of three it's three constituent parts

Reply:
agreed